input.go

```diff
@@ -0,0 +1,19 @@
+package shaping
+
+import (
+	"github.com/go-text/di"
+	"github.com/go-text/font"
+	"golang.org/x/image/math/fixed"
+)
+
+type Input interface {
+	// Text returns the characters to be shaped.
+	Text() []rune
```

Comment:
Harfbuzz supports adding context, that is runes before and after the actual segment, which won't be shaped, but may influence the shaping.
We could support it by changing `Text` to 
`Text() (text []rune, start, length int)`
meaning that the actual text shaped is `text[start:start+length]`, but the whole slice may be used for context information.
You can just return `text, 0, -1` to shape the whole string.
(See [AddRunes](https://pkg.go.dev/github.com/benoitkugler/textlayout@v0.0.1/harfbuzz#Buffer.AddRunes) for more details.)